blocks/components/editable/index.js

```diff
@@ -115,7 +133,16 @@ export default class Editable extends wp.element.Component {
 	componentDidUpdate( prevProps ) {
 		if ( this.props.tagName !== prevProps.tagName ) {
 			this.initialize();
-		} else if ( this.props.value !== prevProps.value ) {
+		}
+
+		if ( this.props.focus !== prevProps.focus && this.props.focus ) {
```

Comment:
Because of the strict comparison of objects and how the reducer is creating next state, this condition passes much more frequently than we'd probably expect. For example I see upon adding a `console.log` to the inner block that it gets called twice every time I select a block.